onnxruntime/core/framework/orttraining_partial_executor.cc

```diff
@@ -233,6 +233,24 @@ Status PartialExecutor::Execute(const SessionState& session_state, const std::ve
     // construct OpKernelContext
     // TODO: log kernel inputs?
     OpKernelContextInternal op_kernel_context(session_state, frame, *p_op_kernel, logger, false);
+
+    // Cache lookup. Currently we only cache single-output nodes,
```

Comment:
>single [](http://example.com/codeflow?start=45&length=6)

Does caching more than one output impact the memory more?

Reply:
It may impact the peak memory usage if the extra outputs cached would not have otherwise been saved for backward. To avoid checking the cache more frequently this is an added restriction. For now, we only cache Cast outputs which follow this rule.